R/submit-model.R

```diff
@@ -200,3 +203,272 @@ submit_nonmem_model <- function(.mod,
 
   return(res)
 }
+
+
+#########################################################
+# S3 dispatches for submitting multiple models in batch
+#########################################################
+
+#' S3 generic method for submit_models
+#' @param .mods generic list or vector of models to submit
+#' @param .bbi_args A named list specifying arguments to pass to babylon formatted like `list("nm_version" = "nm74gf_nmfe", "json" = T, "threads" = 4)`. Run `print_nonmem_args()` to see valid arguments.
+#' @param .mode Either "local" for local execution or "sge" to submit model(s) to the grid
+#' @param ... args passed through to `bbi_exec()`
+#' @param .config_path Optionally specify a path to a babylon.yml config. If not specified, the config in the model directory will be used by default. Path MUST be either an absolute path or relative to the model directory.
+#' @param .wait Boolean for whether to wait for the bbi process to return before this function call returns.
+#' @param .dry_run Returns an object detailing the command that would be run, insted of running it. This is primarily for testing but also a debugging tool.
+#' @param .directory Model directory which `.mod` path is relative to. Defaults to `options('rbabylon.model_directory')`, which can be set globally with `set_model_directory()`. Only used when passing a path for `.mod` instead of a `bbi_{.model_type}_model` object.
+#' @export
+#' @rdname submit_models
+submit_models <- function(
+  .mods,
+  .bbi_args = NULL,
+  .mode = c("sge", "local"),
+  ...,
+  .config_path=NULL,
+  .wait = TRUE,
+  .dry_run=FALSE,
+  .directory = NULL
+) {
+  UseMethod("submit_models")
+}
+
+#' S3 dispatch for submit_model from bbi_nonmem_model
+#' @param .mods a list of S3 object of class `bbi_nonmem_model` to submit
+#' @param .directory This argument is only used when passing a path to `submit_model()`. When `bbi_nonmem_model` object is passed, `.directory` is inferred from the model object.
+#' @importFrom purrr map map_lgl
+#' @export
+#' @rdname submit_models
+submit_models.list <- function( ###### HAS TO CHECK FOR VALID MODEL TYPES and dispatch correctly
+  .mods,
+  .bbi_args = NULL,
+  .mode = c("sge", "local"),
+  ...,
+  .config_path=NULL,
+  .wait = TRUE,
+  .dry_run=FALSE,
+  .directory = NULL
+) {
+
+  if (!is.null(.directory)) {
+    warning(paste(glue("Passed `.directory = {.directory}` to submit_models.bbi_nonmem_model().") ,
+                  "This argument is only valid when passing a path to `submit_models()`.",
+                  "`bbi_nonmem_model` objects were passed, so `.directory` inferred from `.mod${WORKING_DIR}`"))
+  }
+
+  # check that all elements are a model, and that all models are the same type
+  all_models_classes <- map(.mods, function(.x) { intersect(VALID_MOD_CLASSES, class(.x)) })
```

Comment:
to check for a particular class you use `inherits`

`inherits(.x, VALID_MOD_CLASSES)`

Reply:
I changed this a bit, but I still do want to do the intersect here because I need to check that they're all _the same type_ of model, but only that they're all valid model objects.

Reply:
actually it doesn't need either any more because we're using `YAML_MOD_TYPE` instead